src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -453,7 +453,7 @@ public async Task SaveChangesAsync(IList<UnspentOutputs> unspentOutputs, IEnumer
                     this.rewindDataIndexStore.Save(indexItems);
                 }
 
-                this.cachedRewindDataIndex.Add(height,rewindData);
+                this.cachedRewindDataIndex[height] = rewindData;
```

Comment:
I don't think this is correct, we should never override rewind data because if the height is there it means the block was finalized, and if we reorg rewind should be removed from cache

Reply:
Bellow we have this line which should make sure rewind data is never overwritten
` this.cachedRewindDataIndex.Remove(lastRewindDataItem.Key);`